lightgbm/src/main/scala/com/microsoft/azure/synapse/ml/lightgbm/LightGBMRanker.scala

```diff
@@ -93,13 +93,24 @@ class LightGBMRanker(override val uid: String)
     if (getRepartitionByGroupingColumn) {
       val repartitionedDataset = getOptGroupCol match {
         case None => dataset
-        case Some(groupingCol) => {
-          val df = dataset.repartition(new Column(groupingCol)).cache()
-          //force materialization
-          df.count
-          df
-        }
+        case Some(groupingCol) =>
```

Comment:
I recall this logic was modified in this PR from another user for the ranker:
https://github.com/microsoft/SynapseML/pull/778
who had issues with this logic, so I would just be careful here